test/unit/math/gpu/multiply_test.cpp

```diff
@@ -0,0 +1,284 @@
+#include <stan/math/prim/mat.hpp>
+#include <stan/math/gpu/copy.hpp>
+#include <stan/math/gpu/multiply.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <gtest/gtest.h>
+#include <algorithm>
+#ifdef STAN_OPENCL
+
+TEST(MathMatrix, multiply_c_v) {
+  stan::math::vector_d v(3);
+  v << 1, 2, 3;
+  stan::math::matrix_gpu vv(v);
+  vv = stan::math::multiply(vv, 2.0);
+  stan::math::copy(v, vv);
+  EXPECT_FLOAT_EQ(2.0, v(0));
+  EXPECT_FLOAT_EQ(4.0, v(1));
+  EXPECT_FLOAT_EQ(6.0, v(2));
+}
+
+TEST(MathMatrix, multiply_c_rv) {
+  stan::math::row_vector_d rv(3);
+  rv << 1, 2, 3;
+  stan::math::matrix_gpu vv(rv);
+  vv = stan::math::multiply(vv, 2.0);
+  stan::math::copy(rv, vv);
+  EXPECT_FLOAT_EQ(2.0, rv(0));
+  EXPECT_FLOAT_EQ(4.0, rv(1));
+  EXPECT_FLOAT_EQ(6.0, rv(2));
+}
+
+TEST(MathMatrix, multiply_c_m) {
+  stan::math::matrix_d m(2, 3);
+  m << 1, 2, 3, 4, 5, 6;
+  stan::math::matrix_gpu mm(m);
+  mm = stan::math::multiply(mm, 2.0);
+  stan::math::copy(m, mm);
+  EXPECT_FLOAT_EQ(2.0, m(0, 0));
+  EXPECT_FLOAT_EQ(4.0, m(0, 1));
+  EXPECT_FLOAT_EQ(6.0, m(0, 2));
+  EXPECT_FLOAT_EQ(8.0, m(1, 0));
+  EXPECT_FLOAT_EQ(10.0, m(1, 1));
+  EXPECT_FLOAT_EQ(12.0, m(1, 2));
+}
+
+TEST(MathMatrix, multiply_rv_v_exception_pass_vec) {
+  stan::math::row_vector_d rv;
+  stan::math::vector_d v;
+
+  rv.resize(3);
+  v.resize(3);
+  stan::math::matrix_gpu rvv(rv);
+  stan::math::matrix_gpu vv(v);
+  stan::math::matrix_gpu ans_v(1, 1);
+  EXPECT_NO_THROW(ans_v = stan::math::multiply(rvv, vv));
+}
+
+TEST(MathMatrix, multiply_rv_v_exception_pass_empty) {
+  stan::math::row_vector_d rv;
+  stan::math::vector_d v;
+  rv.resize(0);
+  v.resize(0);
+  stan::math::matrix_gpu rvv(rv);
+  stan::math::matrix_gpu vv(v);
+  stan::math::matrix_gpu ans_vv(1, 1);
+  EXPECT_NO_THROW(ans_vv = stan::math::multiply(rvv, vv));
```

Comment:
Why are there so many EXPECT_NO_THROW tests? I typically think of these as something to pop in once you have seen bugs and are coding a regression test for specific bugs. Did someone recommend these?

Reply:
Some of those were added just so that we checked whether anything breaks in edge cases.